pkg/intermediate/aggregate_test.go

```diff
@@ -211,10 +211,10 @@ func TestAggregationProcess(t *testing.T) {
 		aggregationProcess.Stop()
 	}()
 	aggregationProcess.Start()
```

Comment:
Maybe a comment here saying that `Start` of `aggregationProcess` is blocked until the above go routine with `Stop` finishes is better to have. This will give an idea of their usage.
Proper usage would be having `Start` in a go-routine with some external channel. And until the external channel is closed, `Stop` won't be called.